architecture/core/pipeline-service.md

```diff
@@ -57,9 +63,17 @@ Legend:
 
 ### Tekton Pipelines
 
+#### Trusted Artifacts
+
+Tasks in Konflux build pipelines use Trusted Artifacts to securely share files between tasks (see [ADR-0036](../../ADR/0036-trusted-artifacts.md)). This allows users to include custom Tekton Tasks in build pipelines without jeopardizing build integrity. Trusted Artifacts wrap files into archives stored in OCI registries, with checksums recorded as task results to ensure artifacts are not tampered with between tasks.
+
+#### Partner and Custom Tasks
+
+Konflux supports partner-contributed and custom Tekton Tasks in build/test pipelines (see [ADR-0021](../../ADR/0021-partner-tasks.md)). Tasks are validated through CI checks before being accepted into the [build-definitions repository](https://github.com/konflux-ci/build-definitions). Task results follow standardized naming conventions (see [ADR-0030](../../ADR/0030-tekton-results-naming-convention.md)) including `TEST_OUTPUT` for test-like tasks and `SCAN_OUTPUT` for scan-like tasks.
+
 #### appstudio-pipeline Service Account
 
-The service should offer users a service account for running pipelines.
+The service should offer users a service account for running pipelines (see [ADR-0025](../../ADR/0025-appstudio-pipeline-serviceaccount.md)).
```

Comment:
I want to say that the more limited `appstudio-pipelines` service account was created as part of the old Workspace/"KubeSaw" architecture.  There was a more recent change made to the provisioning of these service accounts - how is it handled today?

This can be addressed in a follow-up pull request.